### ISSUE TITLE
flocker-deploy should supply the node hostname when calling flocker-changestate

### ISSUE THREAD

**Issue by wallrj** [ClusterHQ/flocker#283]
`flocker-changestate` needs to know the hostname of the node that it's running on in order to work out which applications should be running.

Instead of the node attempt to work out its own hostname (which may not be the same as the hostname used in the configuration files); `flocker-deploy` will instead supply the hostname as a command line argument to `flocker-changestate`

Original conversation

```
Richard Wall: So about detecting node hostnames....the laptop running flocker-deploy might be using totally different hostnames for the nodes than the node themselves.

Richard Wall: Could we just say that config files must use of IP addresses for now....

Richard Wall: probably not.

Richard Wall: The nodes might be behind NAT and not know their external IP address.

Richard Wall: I seem to remember a lot of code in HybridCLuster for handling this problem.

Richard Wall: ....or could we make the node hostname part of a flocker config file that must be present on each node?

Jean-Paul Calderone: We're going to have flocker-deploy tell the nodes their names.

Richard Wall: Aha.

Richard Wall: Is there a ticket for that?

Richard Wall: Oh. so I suppose that could just be an extra parameter to flocker-changestate

Richard Wall: (extra argument I mean)

Jean-Paul Calderone: Right.

Jean-Paul Calderone: I think #252 is the issue maybe?

Richard Wall: Ok. I'll go work on that part @Adam Dangoor if that's ok with you.

Richard Wall: Maybe I could treat it as a new issue and branch.

Adam Dangoor: yes sorry I thought that's what we said earlier

Jean-Paul Calderone: It seems like it'll just conflict with what Adam's doing?

Adam Dangoor: up to you, it doesnt seem too big to me

Richard Wall: ok.

Jean-Paul Calderone: flocker-deploy needs to change to *pass* the extra argument, maybe that's separate enough to work on in parallel

Richard Wall: Yeah, ok. I'll do that.
```
